streamer/src/sendmmsg.rs

```diff
@@ -76,21 +73,41 @@ fn mmsghdr_for_packet(
     hdr.msg_hdr.msg_iovlen = 1;
     hdr.msg_hdr.msg_name = addr as *mut _ as *mut _;
 
-    #[allow(deprecated)]
-    match InetAddr::from_std(dest) {
-        InetAddr::V4(dest) => {
+    match *dest {
+        SocketAddr::V4(ref socket_addr_v4) => {
             unsafe {
-                std::ptr::write(addr as *mut _ as *mut _, dest);
+                std::ptr::write(
+                    addr as *mut _ as *mut _,
+                    libc::sockaddr_in {
+                        sin_family: libc::AF_INET as u16,
+                        sin_port: socket_addr_v4.port().to_be(),
+                        sin_addr: libc::in_addr {
+                            s_addr: u32::from_ne_bytes(socket_addr_v4.ip().octets()),
+                        },
+                        ..std::mem::zeroed()
+                    },
+                );
             }
             hdr.msg_hdr.msg_namelen = SIZE_OF_SOCKADDR_IN as u32;
         }
-        InetAddr::V6(dest) => {
+        SocketAddr::V6(ref socket_addr_v6) => {
             unsafe {
-                std::ptr::write(addr as *mut _ as *mut _, dest);
+                std::ptr::write(
+                    addr as *mut _ as *mut _,
+                    libc::sockaddr_in6 {
+                        sin6_family: libc::AF_INET6 as u16,
+                        sin6_port: socket_addr_v6.port().to_be(),
+                        sin6_addr: libc::in6_addr {
+                            s6_addr: socket_addr_v6.ip().octets(),
+                        },
+                        sin6_flowinfo: socket_addr_v6.flowinfo(),
+                        sin6_scope_id: socket_addr_v6.scope_id(),
+                    },
+                );
```

Comment:
can you please also add some comments here what is the reference for this implementation?

Reply:
sure! https://github.com/anza-xyz/agave/pull/557/commits/cbd0b39fbf70c1a934979d8fc2464d0c20ebee22

Reply:
sorry for not reading nix lib completely. I found that `SocketAddr` has already implemented in 0.26.4 which is the version we're using. Try to have another solution for this one. [d09277a](https://github.com/anza-xyz/agave/pull/557/commits/d09277acb02f5c108434015eb79d3d2339c58787)
basically I do
1) convert `std::net::SocketAddr` to `nix::sys::socket::SockaddrIn`/`nix::sys::socket::SockaddrIn6`
2) get `libc::sockaddr_in`/`libc::sockaddr_in6` from the value that converted by the previous step


btw, I think I can have another PR for adding a bench test and comparing with socket2.